### PR TITLE
Security scans: Improving telemetry error messages

### PR DESCRIPTION
**Pull request by laileni-aws**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Problem
- Telemetry error messages are not consistent in VSC and JB
## Solution
-  Improving telemetry error messages to keep consistent in both IDE's

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ashishrp-aws — APPROVED** [review]

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt [hunk 1]
```diff
@@ -441,6 +432,17 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
         sleep(CODE_SCAN_POLLING_INTERVAL_IN_SECONDS * TOTAL_MILLIS_IN_SECOND)
     }
 
+    private fun getTelemetryErrorMessage(e: Exception): String = when {
```
Comment (manodnyab) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt:435]: Can we add a test for this function?

**Review by manodnyab — APPROVED** [review]

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt [hunk 2]
```diff
@@ -441,6 +432,17 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
         sleep(CODE_SCAN_POLLING_INTERVAL_IN_SECONDS * TOTAL_MILLIS_IN_SECOND)
     }
 
+    fun getTelemetryErrorMessage(e: Exception): String = when {
+        e.message?.contains("Resource not found.") == true -> "Resource not found."
+        e.message?.contains("Service returned HTTP status code 407") == true -> "Service returned HTTP status code 407"
```
Comment (ashishrp-aws) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt:437]: Thanks for compiling this list.

**Review by ashishrp-aws — APPROVED** [review]

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanException.kt [hunk 3]
```diff
@@ -5,33 +5,35 @@ package software.aws.toolkits.jetbrains.services.codewhisperer.codescan
 
 import software.aws.toolkits.resources.message
 
-open class CodeWhispererCodeScanException(override val message: String?) : RuntimeException()
+open class CodeScanException(override val message: String, open val code: String?) : RuntimeException(message)
```
Comment (manodnyab) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanException.kt:8]: Should there be an enum for the code values possible?

Comment (laileni-aws) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanException.kt:8]: There can be, Let me raise a follow up CR addressing this along with more add on modifications from metrics.

**Pull request merged**